scripts/irods/test/test_resource_types.py

```diff
@@ -2385,6 +2385,153 @@ def test_auto_repl_on__2941(self):
         self.admin.assert_icommand("ils -L " + filename, 'STDOUT_SINGLELINE', 'cacheResc')
         self.admin.assert_icommand("ils -L " + filename, 'STDOUT_SINGLELINE', 'archiveResc')
 
+    def test_get_stale_cache_replica(self):
+
+        try:
+            filename = "testgetstalecachereplica.txt"
+            lib.make_file(filename, 10, 'arbitrary')
+            stale_replica_filename = "stalereplica.txt"
+            lib.make_file(stale_replica_filename, 20, 'arbitrary')
```

Comment:
You can likely avoid these lines by using `istream` to create the data object. For example, you'd replace the calls to `iput` with the following:
```python
self.admin.assert_icommand("istream write " + filename, input="contents of data object")
```

Then you don't need a finally-block.

Just something to consider (not required though).